scripts/format.py

```diff
@@ -120,51 +121,33 @@
 force = False
 
 
-def print_usage():
-    print("Usage: python scripts/format.py [revision|--all] [--check|--fix] [--force]")
-    print(
-        "   [revision]     is an optional revision number, all files that changed since that revision will be formatted (default=HEAD)"
-    )
-    print("                  if [revision] is set to --all, all files will be formatted")
-    print("   --check only prints differences, --fix also fixes the files (--check is default)")
+parser = argparse.ArgumentParser(prog="python scripts/format.py", description='Format source directory files')
+parser.add_argument('--revision', type=str, default='HEAD', help='Revision to format from (default: HEAD)')
+parser.add_argument('--check', action='store_true', help='Only print differences (default)')
+parser.add_argument('--fix', action='store_true', help='Fix the files')
+parser.add_argument("--all", action="store_true", help="Format all files")
+parser.add_argument("--directories", action="store_true", help="Format specified directories")
```

Comment:
Instead of using `rest` for directories, we can use `nargs='*'`, see [here](https://docs.python.org/3/library/argparse.html#usage). This should also allow the previous usage with revision to remain unchanged.